src/main/java/com/snowplowanalytics/snowplow/tracker/emitter/AbstractEmitter.java

```diff
@@ -93,37 +104,49 @@ protected Builder2 self() {
     protected AbstractEmitter(final Builder<?> builder) {
 
         // Precondition checks
-        Preconditions.checkNotNull(builder.httpClientAdapter);
         Preconditions.checkArgument(builder.threadCount > 0, "threadCount must be greater than 0");
 
-        this.httpClientAdapter = builder.httpClientAdapter;
+        if (builder.httpClientAdapter != null) {
+            this.httpClientAdapter = builder.httpClientAdapter;
+        } else {
+            Preconditions.checkNotNull(builder.collectorUrl, "Collector url must be specified if not using a httpClientAdapter");
+
+            // use okhttp as a default
+            OkHttpClient client = new OkHttpClient.Builder()
+                .connectTimeout(5, TimeUnit.SECONDS)
+                .readTimeout(5, TimeUnit.SECONDS)
+                .writeTimeout(5, TimeUnit.SECONDS)
```

Comment:
The defaults in OkHttpClient is 10 seconds for each of these. Just wondering, is there reason you change the defaults, or is it just because 5 seconds feels about right?

Reply:
Not really. The old demo application used to seet them to 5 seconds so I copied and pasted that. You make a good point though, if we're going to use a default OkHttpClient we should leave it as default rather than fiddling with it.